v2/internal/testcommon/test_context.go

```diff
@@ -54,14 +54,28 @@ type PerTestContext struct {
 	Namespace           string
 }
 
-// If you modify this make sure to modify the cleanup-test-azure-resources target in the Makefile too
+// There are two prefixes here because each represents a resource kind with a distinct lifecycle.
```

Comment:
It might be useful to clarify that the former tests are run in an envtest-based cluster while the latter ones are in a real cluster?

Reply:
Added that clarification in the comments onto each `Prefix`